compiler/rustc_lint/src/nonstandard_style.rs

```diff
@@ -366,7 +367,9 @@ impl<'tcx> LateLintPass<'tcx> for NonSnakeCase {
                 })
         };
 
-        if let Some(ident) = &crate_ident {
+        if let Some(ident) = &crate_ident
+            && cx.tcx.crate_types().iter().all(|&crate_type| crate_type != CrateType::Executable)
```

Comment:
This condition can be checked before evaluating `crate_ident`.

Also, it doesn't matter much, but I'd change the condition to `.any(|&crate_type| crate_type != CrateType::Executable)` to still warn on library+executable pairs.

Reply:
I hoisted the check before evaluating `crate_ident`, and made it so if the crate types are mixed library / executable we still check those.
@rustbot ready